lhotse/features/opensmile.py

```diff
@@ -0,0 +1,149 @@
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, List, Union, Optional, Sequence
+
+import numpy as np
+
+from lhotse.features.base import FeatureExtractor, register_extractor
+from lhotse.utils import is_module_available, Seconds, compute_num_frames
+
+@dataclass
+class OpenSmileConfig:
+    """ 
+    OpenSmile configs are stored in separated txt files in its specific format. 
+    You can specify predefined config by setting ``feature_set`` and ``feature_level`` 
+    class attributes with:
+    (1) ``FeatureSet`` and ``FeatureLevel`` classes predefined in
+    https://github.com/audeering/opensmile-python/blob/master/opensmile/core/define.py 
+    OR
+    (2) strings refered to enum members,  
+    You can also create your own config file and pass its path and corresponding feature level 
+    as documented here https://audeering.github.io/opensmile-python/usage.html#custom-config.
+    """
+    feature_set:Union[str, Any] = 'ComParE_2016' # default feature set, string with set name or path to a custom config file
+    feature_level:Union[str, Any] = 'lld' # default feature level or level name if a custom config file is used
+    options: dict = None           # dictionary with optional script parameters
+    loglevel: int = 2              # log level (0-5), the higher the number the more log messages are given
+    logfile: str = None            # if not ``None`` log messages will be stored to this file
+    sampling_rate: int = 16000     # If ``None`` it will call ``process_func`` with the actual sampling rate of the signal.
+    channels: Union[int, Sequence[int]] = 0 # channel selection, see :func:`audresample.remix`
+    mixdown: bool = False          # apply mono mix-down on selection
+    resample: bool = False         # if ``True`` enforces given sampling rate by resampling
+    num_workers: Optional[int] = 1 # number of parallel jobs or 1 for sequential processing. If ``None`` will be set 
+                                   # to the number of processors    
+    verbose: bool = False          # show debug messages        
+                               
+
+    def to_dict(self) -> Dict[str, Any]:
+        return asdict(self)
+    
+    @staticmethod
+    def from_dict(data: Dict[str, Any]) -> "OpenSmileConfig":
+        return OpenSmileConfig(**data)
+    
+    @staticmethod
+    def featuresets_names():
+        """
+        Returns list of strings with names of pretrained FeatureSets available in opensmile.
+        """
+        assert is_module_available("opensmile"), \
+         'To use opensmile extractors, please "pip install opensmile" first.'
+        import opensmile
+        return list(opensmile.FeatureSet.__members__)
+
+
+@register_extractor
+class OpenSmileWrapper(FeatureExtractor):
+    """Wrapper for extraction of features implemented in OpenSmile."""
+
+    name = "opensmile-wrapper"
+    config_type = OpenSmileConfig
+    smileExtractor = None
```

Comment:
I think this line can be removed because you're always setting this field in the constructor.

Reply:
Agree. Done.